src/webgl/p5.Camera.js

```diff
@@ -1638,6 +1638,324 @@ p5.Camera = class Camera {
+        (1 - amt) * m0[7] + amt * m1[7]
+      ).normalize();
+
+      // set the camera
+      // The eye position and center position are calculated based on the front vector.
+      this.camera(
+        lerpedMediumX + ratio * lerpedDist * lerpedFront.x,
+        lerpedMediumY + ratio * lerpedDist * lerpedFront.y,
+        lerpedMediumZ + ratio * lerpedDist * lerpedFront.z,
+        lerpedMediumX + (ratio-1) * lerpedDist * lerpedFront.x,
+        lerpedMediumY + (ratio-1) * lerpedDist * lerpedFront.y,
+        lerpedMediumZ + (ratio-1) * lerpedDist * lerpedFront.z,
+        lerpedUp.x, lerpedUp.y, lerpedUp.z
+      );
+      return;
+    }
+
+    // Calculates the axis vector and the angle of the difference orthogonal matrix.
+    // The axis vector is what I explained earlier in the comments.
+    // similar calculation is here:
+    // https://github.com/mrdoob/three.js/blob/dev/src/math/Quaternion.js#L294
+    let a, b, c, sinTheta;
+    let invOneMinusCosTheta = 1 / (1 - cosTheta);
+    if (m[0] > m[4] && m[0] > m[8]) {
+      a = Math.sqrt((m[0] - cosTheta) * invOneMinusCosTheta); // not zero.
+      invOneMinusCosTheta /= a;
+      b = 0.5 * (m[1] + m[3]) * invOneMinusCosTheta;
+      c = 0.5 * (m[2] + m[6]) * invOneMinusCosTheta;
+      sinTheta = 0.5 * (m[7] - m[5]) / a;
+    } else if (m[4] > m[8]) {
+      b = Math.sqrt((m[4] - cosTheta) * invOneMinusCosTheta); // not zero.
+      invOneMinusCosTheta /= b;
+      c = 0.5 * (m[5] + m[7]) * invOneMinusCosTheta;
+      a = 0.5 * (m[1] + m[3]) * invOneMinusCosTheta;
+      sinTheta = 0.5 * (m[2] - m[6]) / b;
+    } else {
+      c = Math.sqrt((m[8] - cosTheta) * invOneMinusCosTheta); // not zero.
+      invOneMinusCosTheta /= c;
+      a = 0.5 * (m[2] + m[6]) * invOneMinusCosTheta;
+      b = 0.5 * (m[5] + m[7]) * invOneMinusCosTheta;
+      sinTheta = 0.5 * (m[3] - m[1]) / c;
+    }
+
+    // Constructs a new matrix after interpolating the angles.
+    // Multiplying mat0 by the first matrix yields mat1, but by creating a state
+    // in the middle of that matrix, you can obtain a matrix that is
+    // an intermediate state between mat0 and mat1.
+    const angle = amt * Math.atan2(sinTheta, cosTheta);
+    const cosAngle = Math.cos(angle);
+    const sinAngle = Math.sin(angle);
+    const oneMinusCosAngle = 1 - cosAngle;
+    const ab = a * b;
+    const bc = b * c;
+    const ca = c * a;
+    const result = [
+      cosAngle + oneMinusCosAngle * a * a,
+      oneMinusCosAngle * ab - sinAngle * c,
+      oneMinusCosAngle * ca + sinAngle * b,
+      oneMinusCosAngle * ab + sinAngle * c,
+      cosAngle + oneMinusCosAngle * b * b,
+      oneMinusCosAngle * bc - sinAngle * a,
+      oneMinusCosAngle * ca - sinAngle * b,
+      oneMinusCosAngle * bc + sinAngle * a,
+      cosAngle + oneMinusCosAngle * c * c
+    ];
+
+    // Multiply this to mat0 from left to get the interpolated front vector.
+    const newFrontX = result[0] * m0[2] + result[1] * m0[5] + result[2] * m0[8];
```

Comment:
We could maybe add a `multiplyVec3` to `p5.Matrix` to use here (again only if we use p5.Matrix)